lib/bundler/gem_version_promoter.rb

```
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+module Bundler
+  class GemVersionPromoter
+    attr_reader :level, :locked_specs, :unlock_gems
+    attr_accessor :strict, :minimal
+
+    def initialize(locked_specs = SpecSet.new([]), unlock_gems = [])
+      @level_default = :major
+      @level = @level_default
+      @strict = false
+      @minimal = false
+      @locked_specs = locked_specs
+      @unlock_gems = unlock_gems
+      @sort_versions = {}
+    end
+
+    def level=(value)
+      v = begin
+        case value
+        when String, Symbol
+          value.to_sym
+        end
+      end
+
+      @level = [:major, :minor, :patch].include?(v) ? v : @level_default
```

imho this should raise for an unknown level value